tfx/experimental/pipeline_testing/examples/chicago_taxi_pipeline/taxi_pipeline_regression_e2e_test.py

```diff
@@ -0,0 +1,131 @@
+from tfx.experimental.pipeline_testing import stub_component_launcher
+from tfx.orchestration import metadata
+from tfx.orchestration.beam.beam_dag_runner import BeamDagRunner
+from tfx.orchestration.config import pipeline_config
+
+class TaxiPipelineRegressionEndToEndTest(tf.test.TestCase):
+
+  def setUp(self):
+    super(TaxiPipelineRegressionEndToEndTest, self).setUp()
+    self._test_dir = os.path.join(
+        os.environ.get('TEST_UNDECLARED_OUTPUTS_DIR', self.get_temp_dir()),
+        self._testMethodName)
+    self._pipeline_name = 'beam_test'
+    # This example assumes that the taxi data and taxi utility function are stored in
+    # ~/tfx/experimental/pipeline_testing/examples/chicago_taxi_pipeline.
+    # Feel free to customize this as needed.
+    taxi_root = os.path.dirname(__file__)
+    self._data_root = os.path.join(taxi_root, 'data', 'simple')
+    self._module_file = os.path.join(taxi_root, 'taxi_utils.py')
+    self._serving_model_dir = os.path.join(self._test_dir, 'serving_model')
+    self._pipeline_root = os.path.join(self._test_dir, 'tfx', 'pipelines',
+                                       self._pipeline_name)
+    # Metadata path for recording successful pipeline run.
+    self._metadata_db_uri = os.path.join(self._test_dir, 'tfx', 'record',
+                                         'metadata.db')
+    # Metadata path for stub pipeline runs.
+    self._metadata_path = os.path.join(self._test_dir, 'tfx', 'metadata',
+                                       self._pipeline_name, 'metadata.db')
+    self._output_dir = os.path.join(self._test_dir, 'testdata')
+
+  def testTaxiPipelineBeam(self):
+    # Runs the pipeline and record to self._output_dir
+    record_taxi_pipeline = taxi_pipeline_beam._create_pipeline(  # pylint:disable=protected-access, unexpected-keyword-arg
+        pipeline_name=self._pipeline_name,
+        data_root=self._data_root,
+        module_file=self._module_file,
+        serving_model_dir=self._serving_model_dir,
+        pipeline_root=self._pipeline_root,
+        metadata_path=self._metadata_db_uri,
+        beam_pipeline_args=[])
+    BeamDagRunner().run(record_taxi_pipeline)
+    pipeline_recorder_utils.record_pipeline(
+        output_dir=self._output_dir,
+        metadata_db_uri=self._metadata_db_uri,
+        host=None,
+        port=None,
+        pipeline_name=self._pipeline_name,
+        run_id=None)
+
+    # Run pipeline with stub executors first time.
+    taxi_pipeline = taxi_pipeline_beam._create_pipeline(  # pylint:disable=protected-access, unexpected-keyword-arg
+        pipeline_name=self._pipeline_name,
+        data_root=self._data_root,
+        module_file=self._module_file,
+        serving_model_dir=self._serving_model_dir,
+        pipeline_root=self._pipeline_root,
+        metadata_path=self._metadata_path,
+        beam_pipeline_args=[])
+
+    model_resolver_id = 'ResolverNode.latest_blessed_model_resolver'
+    self._component_ids = [component.id
```

Comment:
Should we store this as a property of self? It seems unnecessary in this case because the value is meaningful only in this method.